gossipd/gossip.c

```diff
@@ -623,6 +623,17 @@ static void send_node_announcement(struct daemon *daemon)
 			      tal_hex(tmpctx, err));
 }
 
+/* Should we announce our own node? */
+static void consider_own_node_announce(struct daemon *daemon)
```

Comment:
Agreed, renamed to "maybe_send_own_node_announce".

We *could* look at our own node's channels before and after in each case, but I was trying to avoid that originally.  Ideally, you can call maybe_send_own_announce() from anywhere, and it Does The Right Thing.  Turns out we don't need it in too many places, though.

Reply:
Oh right, I didn't catch that this would be a loop over all channels.